osquery/events/windows/etw/etw_publisher.cpp

```diff
@@ -34,4 +35,80 @@ EtwPublisherBase::getPostProcessorCallback() {
   };
 }
 
+void EtwPublisherBase::updateUserInfo(const std::string& userSid,
+                                      std::string& username) {
+  // Updating user information using gathered user SIDs as input
+  auto usernameIt = usernamesBySIDs_.find(userSid);
+  if (usernameIt != usernamesBySIDs_.end()) {
+    auto cachedUsername = usernameIt->second;
+    username.assign(cachedUsername);
+  } else {
+    PSID pSid = nullptr;
+
+    if (!ConvertStringSidToSidA(userSid.c_str(), &pSid) || pSid == nullptr) {
```

Comment:
Use unicode version

Reply:
Addressing this in a separate PR as it will affect both the existing process events table and this one.